daml-script/daml3/Daml/Script/Questions/Submit/Error.daml

```diff
@@ -15,12 +16,26 @@ data DevErrorType
   | UnknownNewFeature -- ^ This should never happen - Update Scripts when you see this!
   deriving Show
 
+-- | Additional debugging info optionally provided by the ledger
+data ContractNotFoundAdditionalInfo
+  = NotFound -- The generic "this contract does not exist"
+  | NotActive -- The contract did exist, but has since been archived
+  | NotEffective with -- The contract "will" exist, only thrown from the IDELedger where one can go backwards in time
+      effectiveAt : Text
+  | NotVisible with -- The contract exists, but the readAs/actAs party union cannot see it
+      actAs : [Party]
+      readAs : [Party]
+      observers : [Party]
+  deriving Show
+
```

Comment:
Could we hide that to the final user, in order to avoid he relies on it  ?

Reply:
@remyhaemmerle-da What do you think of latest changes in https://github.com/digital-asset/daml/pull/17211 ? We added an `isNotActive` function, which I'm not terribly happy about - we can build  a checking function for each variant of `ContractNotFoundAdditionalInfo`

Reply:
Currently we use it to properly test https://github.com/digital-asset/daml/pull/17211/files#diff-a2ffb6523c674b6a7bb55198f804639fc8427dfaf33c6c5a74947c6d8c8bcc89R104-R114

Reply:
This does not change the root of my concern. This prevents us to change this error in the future. 
We have to find a way to solve that, but  we can postpone it for latter.